python/ray/tune/suggest/hebo.py

```diff
@@ -0,0 +1,359 @@
+import logging
+import pickle
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from ray.tune.result import DEFAULT_METRIC
+from ray.tune.sample import Categorical, Domain, Float, Integer, LogUniform, \
+    Quantized, \
+    Uniform
+from ray.tune.suggest.suggestion import UNRESOLVED_SEARCH_SPACE, \
+    UNDEFINED_METRIC_MODE, UNDEFINED_SEARCH_SPACE
+from ray.tune.suggest.variant_generator import parse_spec_vars
+from ray.tune.utils.util import is_nan_or_inf, unflatten_dict
+
+try:  # Python 3 only -- needed for lint test.
+    import bo as hebo
+    import torch  # hebo has torch as a dependency
+except ImportError:
+    hebo = None
+
+from ray.tune.suggest import Searcher
+
+logger = logging.getLogger(__name__)
+
+SPACE_ERROR_MESSAGE = ("Space must be either a HEBO DesignSpace object"
+                       "or a dictionary with ONLY tune search spaces.")
+
+
+def _validate_warmstart(parameter_names: List[str],
+                        points_to_evaluate: List[Union[List, Dict]],
+                        evaluated_rewards: List):
+    if points_to_evaluate:
+        if not isinstance(points_to_evaluate, list):
+            raise TypeError(
+                "points_to_evaluate expected to be a list, got {}.".format(
+                    type(points_to_evaluate)))
+        for point in points_to_evaluate:
+            if not isinstance(point, (dict, list)):
+                raise TypeError(
+                    f"points_to_evaluate expected to include list or dict, "
+                    f"got {point}.")
+
+            if not len(point) == len(parameter_names):
+                raise ValueError("Dim of point {}".format(point) +
+                                 " and parameter_names {}".format(
+                                     parameter_names) + " do not match.")
+
+    if points_to_evaluate and evaluated_rewards:
+        if not isinstance(evaluated_rewards, list):
+            raise TypeError(
+                "evaluated_rewards expected to be a list, got {}.".format(
+                    type(evaluated_rewards)))
+        if not len(evaluated_rewards) == len(points_to_evaluate):
+            raise ValueError(
+                "Dim of evaluated_rewards {}".format(evaluated_rewards) +
+                " and points_to_evaluate {}".format(points_to_evaluate) +
+                " do not match.")
```

Comment:
This looks similar (identical?) to the one we use in skopt. Should we just move this either to `utils` or to the base searcher class and use the same function then? We might want to use this for other searchers in the futures as well.

Reply:
It is indeed identical. We could make it a part of the Searcher class, I think.

Reply:
@krfricke Should I make this change in this PR?

Reply:
Yep, that would be great!